internal/app/machined/pkg/system/services/networkd.go

```diff
@@ -120,11 +123,27 @@ func (n *Networkd) Runner(config runtime.Configurator) (runner.Runner, error) {
 // HealthFunc implements the HealthcheckedService interface
 func (n *Networkd) HealthFunc(runtime.Configurator) health.Check {
 	return func(ctx context.Context) error {
-		conn, err := grpc.Dial("unix:"+constants.NetworkSocketPath, grpc.WithInsecure())
-		if err != nil {
+		var (
+			conn   *grpc.ClientConn
+			err    error
+			hcResp *healthapi.HealthCheckResponse
+		)
+
+		if conn, err = grpc.Dial("unix:"+constants.NetworkSocketPath, grpc.WithInsecure()); err != nil {
+			return err
+		}
+		defer conn.Close() //nolint: errcheck
+
+		nClient := healthapi.NewHealthClient(conn)
+		if hcResp, err = nClient.Check(ctx, &empty.Empty{}); err != nil {
```

Comment:
should we depend on `Ready()` here if we want to wait for networkd to complete all the initial config?

Reply:
I think we will want to do a combination of both.

If !networkd.Ready() then there's no use in trying a health check.

If networkd.Ready() && networkd.Check() then healthy.

Reply:
Is this in the follow up PR?

Reply:
negative, this is here.
L139-145 - ready check
L147-153 - health check

the health check is 'dumb' right now in that it always returns that it's up/available. This should maintain current functionality.